File: src/IsbInterop/IsbInteropException.vb
''' <summary>
''' Исключение взаимодействия с IS-Builder.
''' </summary>
Public Class IsbInteropException
    Inherits Exception

    ''' <summary>
    ''' Конструктор.
    ''' </summary>
    ''' <param name="message">Сообщение.</param>
    Public Sub New(message As String)
        MyBase.New(message)
    End Sub

    ''' <summary>
    ''' Конструктор.
    ''' </summary>
    ''' <param name="message">Сообщение.</param>
    ''' <param name="innerException">Внутреннее исключение.</param>
    Public Sub New(message As String, innerException As Exception)
        MyBase.New(message, innerException)
    End Sub

End Class

File: src/IsbInterop/IsbInteropException.cs
using System;

namespace IsbInterop
{
  /// <summary>
  /// Исключение взаимодействия с IS-Builder.
  /// </summary>
  public class IsbInteropException : Exception
  {
    /// <summary>
    /// Конструктор.
    /// </summary>
    /// <param name="message">Сообщение.</param>
    public IsbInteropException(string message) : base(message)
    {
    }

    /// <summary>
    /// Конструктор.
    /// </summary>
    /// <param name="message">Сообщение.</param>
    /// <param name="innerException">Внутреннее исключение.</param>
    public IsbInteropException(string message, Exception innerException)
      : base(message, innerException)
    {
    }
  }
}
